src/test/java/com/teragrep/pth_06/planner/walker/conditions/StreamingConditionTest.java

```diff
@@ -0,0 +1,159 @@
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth_06.planner.walker.conditions;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+import org.jooq.Condition;
+import org.jooq.impl.DSL;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class StreamingConditionTest {
+
+    @Test
+    public void testValidTags() {
+        String[] tags = {
+                "index", "sourcetype", "host"
+        };
+        String value = "123456";
+        String operation = "EQUALS";
+        List<Condition> conditions = new ArrayList<>();
+        int loops = 0;
+        for (String tag : tags) {
+            Condition result = Assertions
+                    .assertDoesNotThrow(() -> new StreamingCondition(tag, value, operation).condition());
+            Assertions.assertNotEquals(DSL.noCondition(), result);
+            conditions.add(result);
+            loops++;
+        }
+        Assertions.assertEquals(3, loops);
+        List<Condition> expectedResults = new ArrayList<>();
+        expectedResults.add(new IndexCondition(value, operation, true).condition());
+        expectedResults.add(new SourceTypeCondition(value, operation, true).condition());
+        expectedResults.add(new HostCondition(value, operation, true).condition());
+        Assertions.assertEquals(expectedResults, conditions);
+    }
+
+    @Test
+    public void testStreamingQueryPassThroughTags() {
+        String[] tags = {
+                "earliest", "latest", "index_earliest", "index_latest", "indexstatement"
+        };
+        String value = "123456";
+        String operation = "EQUALS";
+        int loops = 0;
+        for (String tag : tags) {
+            Condition result = Assertions
+                    .assertDoesNotThrow(() -> new StreamingCondition(tag, value, operation).condition());
+            Assertions.assertEquals(DSL.noCondition(), result);
+            loops++;
+        }
+        Assertions.assertEquals(5, loops);
+    }
+
+    @Test
+    public void testUnsupportedElementName() {
+        String tag = "unsupported_tag";
+        String value = "123456";
+        String operation = "EQUALS";
+        IllegalArgumentException ex = Assertions
+                .assertThrows(IllegalArgumentException.class, () -> new StreamingCondition(tag, value, operation).condition());
+        String expectedMessage = "Unsupported element tag <unsupported_tag>";
+        Assertions.assertEquals(expectedMessage, ex.getMessage());
+    }
+
+    @Test
+    public void testEquality() {
+        String tag = "index";
+        String value = "value";
+        String operation = "EQUALS";
```

Comment:
I think the separate equality/non-equality/hashcode tests are not necessary when using EqualsVerifier, as far as I know

Reply:
perhaps a bit unnecessary but no harm keeping them now I guess.